src/gpu/generic/sycl/ref_binary.hpp

```diff
@@ -87,6 +87,8 @@ struct ref_binary_t : public gpu::generic::sycl::primitive_t {
             const auto &scales = attr()->scales_;
             bool dt_ok = true;
             for (auto arg : supported_args) {
+                if (scales.has_default_values(arg)) continue;
+
```

Comment:
It might also be worth checking that unsupported args for scales are set to default with 

scales.has_default_values(supported_args); (see [here](https://github.com/oneapi-src/oneDNN/blob/fe85475acb78db87cf2798c93353411760673474/src/common/primitive_attr_quant.hpp#L156))

Reply:
There are two versions of has_default_values:
- one that take arg_kind, and checks that the scale for that arg_kind is default
- one that takes an std::vector of supported args, and checks that unsupported args are default.

IMHO, it would be safer here to rely on the second flavor and call `scales.has_default_values(supported_args)` outside of the loop.

Reply:
@mgouicem does this make sense? Or would you still like me to make a change here?

Reply:
@sgeor255, we need 1 things to be checked:
- that all scales set by user are supported by the implementation. Calling `scales.has_default_values(supported_args)` outside of the loop checks that.
- that all scales supported by the implementation are properly set. The check you currently have inside the loop does that (you are checking if scales are not default before accessing data_type).

Here we would need both and it seems the first one is missing.

Reply:
I see, thanks, updated it as suggested.